ocrd/cli/zip.py

```diff
@@ -0,0 +1,105 @@
+@click.option('-Z', '--skip-zip', help="Create a directory but do not ZIP it", is_flag=True, default=False)
+@click.option('-j', '--processes', help="Number of parallel processes", type=int, default=1)
+def bag(directory, mets_basename, dest, identifier, in_place, manifestation_depth, mets, base_version_checksum, tag_file, skip_zip, processes):
+    """
+    Bag workspace as OCRD-ZIP at DEST
+    """
+    resolver = Resolver()
+    workspace = Workspace(resolver, directory=directory, mets_basename=mets_basename)
+    workspace_bagger = WorkspaceBagger(resolver)
+    workspace_bagger.bag(
+        workspace,
+        dest=dest,
+        ocrd_identifier=identifier,
+        ocrd_manifestation_depth=manifestation_depth,
+        ocrd_mets=mets,
+        ocrd_base_version_checksum=base_version_checksum,
+        processes=processes,
+        tag_files=tag_file,
+        skip_zip=skip_zip,
+        in_place=in_place
+    )
+
+# ----------------------------------------------------------------------
+# ocrd zip spill
+# ----------------------------------------------------------------------
+
+@zip_cli.command('spill')
+@click.option('-d', '--directory',
+              default='.',
+              type=click.Path(file_okay=False, dir_okay=True, writable=True, resolve_path=True),
+              help='Workspace folder location.',
+              show_default=True)
+@click.argument('src', type=click.Path(dir_okay=False, readable=True, resolve_path=True), required=True)
+def spill(directory, src):
+    """
+    Spill/unpack OCRD-ZIP bag at SRC to DEST
+
+    SRC must exist an be an OCRD-ZIP
+    DEST must not exist and be a directory
+    """
+    resolver = Resolver()
+    workspace_bagger = WorkspaceBagger(resolver)
+    workspace = workspace_bagger.spill(src, directory)
+    print(workspace)
+
+# ----------------------------------------------------------------------
+# ocrd zip validate
+# ----------------------------------------------------------------------
+
+@zip_cli.command('validate')
+@click.argument('src', type=click.Path(dir_okay=True, readable=True, resolve_path=True), required=True)
+@click.option('-Z', '--skip-unzip', help="Treat SRC as a directory not a ZIP", is_flag=True, default=False)
+@click.option('-B', '--skip-bag', help="Whether to skip all checks of manifests and files", is_flag=True, default=False)
+@click.option('-C', '--skip-checksums', help="Whether to omit checksum checks but still check basic BagIt conformance", is_flag=True, default=False)
+@click.option('-W', '--skip-workspace', help="Whether to skip the workspace check of files in /data", is_flag=True, default=False)
+@click.option('-D', '--skip-delete', help="Whether to skip deleting the unpacked OCRD-ZIP dir after valdiation", is_flag=True, default=False)
+@click.option('-j', '--processes', help="Number of parallel processes", type=int, default=1)
+def validate(src, **kwargs):
+    """
+    Validate OCRD-ZIP
+
+    SRC must exist an be an OCRD-ZIP
```

Comment:
Update comment. (`--skip-unizp` says that SRC may be a directory)

Reply:
Is an unzipped OCRD-ZIP not still an OCRD-ZIP? Maybe

`be an OCRD-ZIP or, either a ZIP file or a directory`?

Reply:
Perfect!